src/uct/tcp/tcp_sockcm.c

```diff
@@ -60,15 +60,17 @@ static inline void uct_tcp_sockcm_ep_handle_event_status(uct_tcp_sockcm_ep_t *ep
                                                          ucs_status_t status,
                                                          int events, const char *reason)
 {
-    ucs_assert(!(ep->state & UCT_TCP_SOCKCM_EP_FAILED));
+    if (status != UCS_OK) {
```

Comment:
can it ever be OK?

Reply:
yes, uct_tcp_sockcm_ep_handle_event_status is called in any case. maybe then i'll call it only if status isn't OK...

Reply:
seems it's only called from uct_tcp_sa_data_handler() which takes return value from uct_tcp_sockcm_event_err_to_ucs_err_log() which is never UCS_OK

Reply:
it's called from all the events (EVREAD/EVWRITE), not only for EVERR

Reply:
ok

Reply:
so i'll call uct_tcp_sockcm_ep_handle_event_status in all places only if status isn't UCS_OK

Reply:
ok